pkg/model/message.go

```diff
@@ -14,6 +15,83 @@ type Messages struct {
 	Original bool
 }
 
+type MessagesSlice []Messages
+
+// HasLanguage checks if MessagesSlice contains Messages with the given language.
+func (ms MessagesSlice) HasLanguage(lang language.Tag) bool {
+	return ms.LanguageIndex(lang) != -1
+}
+
+// LanguageIndex returns index of Messages with the given language. If not found, returns -1.
+func (ms MessagesSlice) LanguageIndex(lang language.Tag) int {
+	return slices.IndexFunc(ms, func(m Messages) bool {
+		return m.Language == lang
+	})
+}
+
+// Replace replaces Messages with the same language. If not found, appends to the slice.
+func (ms *MessagesSlice) Replace(messages Messages) {
+	switch idx := ms.LanguageIndex(messages.Language); idx {
+	case -1:
+		*ms = append(*ms, messages)
+	default:
+		(*ms)[idx] = messages
+	}
+}
+
+// SplitOriginal returns a pointer to the original and other messages.
+func (m MessagesSlice) SplitOriginal() (original *Messages, others MessagesSlice) {
+	others = make(MessagesSlice, 0, len(m))
+
+	for i := range m {
+		if m[i].Original {
+			original = &m[i]
+		} else {
+			others = append(others, m[i])
+		}
+	}
+
+	return
+}
+
+/*
+AlterTranslations changes the status of specific messages in all languages to UNTRANSLATED
+based on a list of message IDs, while keeping the original language messages status.
+
+Example:
+
+	untranslatedIDs := { "1" }
+
+	{ Language: en, Original: true, Messages: [ { ID: "1", Message: "Hello", Status: Translated  } ], ...
+	{ Language: fr, Messages: [ { ID: "1", Message: "Bonjour", Status: Translated  } ], ... ] }
+	{ Language: de, Messages: [ { ID: "1", Message: "Hallo", Status: Translated  } ], ... ]
+
+	Result:
+	{ Language: en, Original: true, Messages: [ { ID: "1", Message: "Hello", Status: Translated  } ], ...
+	{ Language: fr, Messages: [ { ID: "1", Message: "Bonjour", Status: Untranslated  }, ... ] }
+	{ Language: de, Messages: [ { ID: "1", Message: "Hallo", Status: Untranslated  } ], ... ]
+*/
+func (ms MessagesSlice) AlterTranslations(untranslatedIDs []string) {
+	if len(untranslatedIDs) == 0 || len(ms) == 1 {
```

Comment:
What is use case for `len(ms) == 1`?

Reply:
Nothing to alter, e.g.
Slice contains only one element, allegedly original -> original files should remain untouched.

But a more straightforward check could be implemented

Reply:
By reading the code, the reader does not know it. Add the moment?